src/common/components/repository-row/dropdowns/register-name-dropdown.js

```diff
@@ -47,6 +54,60 @@ const Caption = (props) => {
   );
 
   let caption;
+  let changeForm;
+  let message;
+
+  const changeRegisteredName = !!registeredName;
+
+  // only show "change name" info and form if there is a name registered already,
+  // user is authenticated and we are not showing register success message
+  const showChangeForm = (
+    changeRegisteredName &&
+    authStore.authenticated &&
+    !registerNameStatus.success
+  );
+
+  if (!changeRegisteredName) {
+    // if user is registering new name
+    message = 'To publish to the snap store, this repo needs a registered name.';
+    if (!authStore.authenticated) {
+      message += ' You need to sign in to Ubuntu One to register a name.';
+    }
+  } else {
+    // if user is changing already registered name
+    if (!authStore.authenticated) {
+      message = 'You need to sign in to Ubuntu One to change a snap’s registered name.';
+    }
+  }
+
+  if (showChangeForm) {
```

Comment:
It's a bit messy now.

This dropdown now needs to support both name registering and changing name (which have different UIs) and signing to the SSO (with 2 different messages depending on the state).

This means a lot of conditionals here...

There is no much to do with this until we have more consistent UI for registering and changing name, which may come later.